### PR TITLE
(PDB-2527) allow latest-report? to be used in legacy events queries

### PR DESCRIPTION
**Pull request by wkalt**
* adds an index on certnames(latest_report_id)
* fixes a bug preventing latest_report? usage with the distinct_resources parameter
* moves query logic for unioned aggregate-event-counts queries from the
  aggregations to the CTE

### CODE REVIEW

**Comment by wkalt** [comment]
ping @kbrezina

File: test/puppetlabs/puppetdb/http/aggregate_event_counts_test.clj [hunk 1]
```diff
@@ -91,7 +91,48 @@
...
+         ["<=" "run_end_time" current-time-str]))
+
+  (store-example-report! (:basic2 reports) (coerce/to-string (now)))
+
+  (are [result query] (= result
+                         (query-result method endpoint query
+                                       {:summarize_by "resource"
+                                        :distinct_resources true
+                                        :distinct_start_time 0
+                                        :distinct_end_time (now)}))
```
Comment (kbrezina) [inline test/puppetlabs/puppetdb/http/aggregate_event_counts_test.clj]: Do `:distinct_start_time` and `:distinct_end_time` parameters need to be provided even if `["=" "latest_report?" true]` query is used?

Comment (wkalt) [inline test/puppetlabs/puppetdb/http/aggregate_event_counts_test.clj]: @kbrezina yes

File: src/puppetlabs/puppetdb/query/events.clj [hunk 2]
```diff
@@ -113,10 +122,8 @@
   (let [where-clause (if where (format "WHERE %s" where) "")]
```
Comment (kbarber) [inline src/puppetlabs/puppetdb/query/events.clj]: where-clause is no longer used

File: test/puppetlabs/puppetdb/http/aggregate_event_counts_test.clj (hunk 1 above)
Comment (dmcpl) [inline test/puppetlabs/puppetdb/http/aggregate_event_counts_test.clj]: Hi @wkalt , if we are going to have to specify start/end times for latest_report? query, is it acceptable then just to use `0` and `now` as the time range as you have done here without fear of fiery explosion?

Comment (wkalt) [inline test/puppetlabs/puppetdb/http/aggregate_event_counts_test.clj]: @dmcpl absolutely, assuming you mean on the client side.

**Comment by ajroetker** [comment]
:+1:

**Pull request merged**